apps/portal/src/app/views/notifications/notifications.component.ts

```diff
@@ -225,6 +228,84 @@ export class NotificationsComponent implements OnInit {
     }
   }
 
+  getProvidersForSelectedApplication() {
+    // Set the query variables
+    const variables = {
+      filters: [],
+      offset: 0,
+      limit: 15,
+    };
+
+    // Set query filters
+    variables.filters.push({
+      field: 'applicationId',
+      operator: 'eq',
+      value: this.selectedApplication.toString(),
+    });
+
+    // Fetch the login token
+    const loginToken = this.getJWTLoginToken();
+
+    if (!loginToken) {
+      // Handle missing token
+      return;
+    }
+
+    // Fetch providers and handle errors
+    this.providerService
+      .getProviders(variables, loginToken)
+      .pipe(
+        // catchError operator to handle errors
+        catchError((error) => {
```

Comment:
We don't have common code for this code block? Can add a TODO task for this

Reply:
Should we create an issue?

Reply:
@kshitij-k-osmosys  No need, directly create a task in pinestem for now in backlog.